src/vm/builtins/builtin_runner/keccak.zig

```diff
@@ -837,6 +837,12 @@ test "KeccakBuiltinRunner: finalStack should return TypeMismatchNotRelocatable e
     var memory_segment_manager = try MemorySegmentManager.init(std.testing.allocator);
     defer memory_segment_manager.deinit();
 
+    try memory_segment_manager.memory.data.append(std.ArrayListUnmanaged(?MemoryCell){});
+    try memory_segment_manager.memory.data.append(std.ArrayListUnmanaged(?MemoryCell){});
+    try memory_segment_manager.memory.data.append(std.ArrayListUnmanaged(?MemoryCell){});
```

Comment:
If I understand correctly, here you add empty `ArrayListUnmanaged` because you have to add 3 segments manually to have access to index 2.

If that's it, in my opinion, the logic for adding empty arrays must be found directly in the `addSegment` function: here it's a little different from Rust, when adding a segment, we can't just add 1 to the number of segments. We must also allocate an empty array there.